x-pack/plugins/uptime/server/lib/requests/search/query_context.ts

```diff
@@ -146,4 +147,21 @@ export class QueryContext {
       this.statusFilter
     );
   }
+
+  // Returns true if the order returned by the ES query matches the requested sort order.
+  // This useful to determine if the results need to be reversed from their ES results order.
+  // I.E. when navigating backwards using prevPagePagination (CursorDirection.Before) yet using a SortOrder.ASC.
+  searchSortAligned(): boolean {
+    if (this.pagination.cursorDirection === CursorDirection.AFTER) {
+      return this.pagination.sortOrder === SortOrder.ASC;
```

Comment:
what changes sortOrder? to me it seems like it always remains same.

Reply:
You're right, nothing does today. Normally I'd say YAGNI, but in this case it helps make the reasoning about this code more clear IMHO, showing that we've thought about sorting. Also, I think we may add, if nothing else, the option to sort by `monitor.id` descending in the future.

Reply:
I'll add normally, adding code now to support future features is not worth it on its own, but when combined with the clarifying nature of this code it makes sense IMHO. That said, I could be persuaded to go the other way